docs: add faqs section to npx docs [TREX-1869] (#486)
snyk/user-docs: docs/scan-with-snyk/pull-requests/pull-request-checks/pull-request-experience.md

## Frequently Asked Questions (FAQs)

### How can false positives be handled?

If a finding is a false positive, the comment can be manually resolved in the SCM. It will remain resolved even if the issue is detected again in a subsequent PR check.

### When a new commit is pushed, what is the behavior of the comments?

#### Issue Summary Comment

When a new commit is pushed to the PR, the existing summary comment is updated with the Snyk results for that commit i.e. the summary with the new count of findings. No additional comment is posted. This provides a clear overview of potential vulnerabilities without overwhelming developers with excessive notifications.

#### Inline Comments

For each commit, a review is created if there are any new findings. Each finding is added as an inline comment. If a finding from a previous commit is no longer present in the new commit, its corresponding inline comment is marked as resolved. Findings that continue to exist remain unchanged as inline comments.

> ### Why can’t I see some findings as comments?

Reviewer: Can we move this question under [this one](https://github.com/snyk/user-docs/pull/486/files#diff-cbc2429ddc3234a82e429d24088ef18a93ebb6712a019ce45fc61161535c23a1R126) and rephrase it to `Why are some findings not appearing as inline comments?`

> Inline comment support varies depending on the SCM platform you are using. The availability of comments depends on where the finding is located in relation to the code changes in your pull request:
> 
> - **GitHub & Bitbucket**
>    - ✅ Supported **inside the diff**
>    - ✅ Supported **outside the diff but within modified files**
>    - ❌ _Not supported outside the diff and modified files_
> 
> - **GitLab & Azure Repos**
>    - ❌ _Inline comments are not supported_
> 
> If you don't see certain findings as comments, it may be due to SCM limitations on where inline comments can appear.

Reviewer: This is not accurate 😅

```suggestion
Inline comments are only available for issues detected by Snyk Code PR Checks. Their behavior may vary depending on the SCM platform. On most platforms, an inline comment is added for each issue found. However, on GitLab, inline comments are only added for issues found within the files that were modified in the pull request.
```

Author: Apologies - I had verified this with Ildi but I may have expressed myself wrongly 😅

> ### Why is my repository not receiving any PR comments?

Reviewer: I believe we can either delete this question since it addresses the same as described on [prerequisites section](https://docs.snyk.io/scan-with-snyk/pull-requests/pull-request-checks/pull-request-experience#prerequisites-for-the-snyk-pull-request-experience) or refer the response to the [section](https://docs.snyk.io/scan-with-snyk/pull-requests/pull-request-checks/pull-request-experience#prerequisites-for-the-snyk-pull-request-experience).

```suggestion
### Why is my repository not receiving any PR comments?

To ensure your repository receives PR comments, please review the [prerequisites section](_link_to_section) for the Snyk Pull Request Experience. Make sure all required conditions are met and properly fulfilled.
```

Reviewer: Also rephrase this to `Why are PR comments not appearing in my repository?`

- If your repository is not receiving PR comments, ensure that everything is configured correctly as per the documentation.

- For the **GitHub integration**, please ensure that the GitHub account and **Personal Access Token (PAT)** for opening fix and upgrade PRs is configured. Check [this](../snyk-pull-or-merge-requests/opening-fix-and-upgrade-pull-requests-from-a-fixed-github-account.md) guide for more details.

<figure><img src="../../../.gitbook/assets/set_pat_fix_upgrade_prs.png" alt=""><figcaption><p>Configuration details to set the PAT for fix and upgrade PRs</p></figcaption></figure>

- Ensure PR checks are working, if not, check [how to troubleshoot PR checks](../pull-request-checks/troubleshoot-pr-checks.md).

> ### Why is my repository not receiving any SAST inline comments?

Reviewer: Rephrase to
```suggestion
### Why are inline comments not appearing in my repository?
```

> If your repository is not receiving SAST inline comments, ensure the following settings are correctly configured:
> 
> 1. Make sure **Snyk Code** is enabled for your organization.
> 
> <figure><img src="../../../.gitbook/assets/enable_snyk_code.png" alt=""><figcaption><p>Configuration details to enable Snyk Code</p></figcaption></figure>
> 
> 2. Navigate to the integration settings and enable **Code Analysis** in the **Pull Request Checks** section.
> 
> <figure><img src="../../../.gitbook/assets/enable_code_analysis.png" alt=""><figcaption><p>Configuration details to enable Code Analysis</p></figcaption></figure>
> 
> 3. In the **Pull Request Experience** section, turn on **Inline Comments** to ensure SAST findings appear directly in the code review.
> 
> <figure><img src="../../../.gitbook/assets/Screenshot 2024-11-27 at 11.44.50.png" alt=""><figcaption><p>Configuration details to enable Inline Comments</p></figcaption></figure>
> 
> ⚠️ If you are using **Snyk Local Code Engine**, inline comments are **not** available for Code Analysis using this engine.

Reviewer: ```suggestion
Inline comments are available only for issues detected by Snyk Code PR Checks. To ensure that issues appear as inline comments, verify the following requirements:

1. **Snyk Code** is enabled for your organization.

<figure><img src="../../../.gitbook/assets/enable_snyk_code.png" alt=""><figcaption><p>Configuration details to enable Snyk Code</p></figcaption></figure>

2. **Code Analysis** under the **Pull request status checks** section is enabled for your integration.

<figure><img src="../../../.gitbook/assets/enable_code_analysis.png" alt=""><figcaption><p>Configuration details to enable Code Analysis</p></figcaption></figure>

3. **Inline Comments** under In the **Pull Request Experience** section is enabled for you integration.

<figure><img src="../../../.gitbook/assets/Screenshot 2024-11-27 at 11.44.50.png" alt=""><figcaption><p>Configuration details to enable Inline Comments</p></figcaption></figure>

⚠️ **Note**: Inline comments are not available if you are using the **Snyk Local Code Engine**.
```

### Is Broker supported?

> Yes, brokered integrations are supported for both **Issue Summary Comment** and **Inline Comments**. Both Legacy and Universal Broker are supported. Note: The broker's version 4.194 and above are supported. 

Reviewer: ```suggestion
Yes. Brokered integrations are supported for both **Issue Summary Comment** and **Inline Comments**, with compatibility for both Legacy and Universal Broker. Note that broker version 4.194 or higher is required.
```
